web3/providers/base.py

```diff
@@ -52,6 +57,28 @@
     from web3 import Web3  # noqa: F401
 
 
+class SupportsBatching(Protocol):
+    _is_batching: bool
```

Comment:
See my comment above, we shouldn't define this twice. Was thinking we should move this into `web3._utils.batching.py` along with the async version.